src/lib/formats/dmk_dsk.cpp

```diff
@@ -83,6 +81,27 @@ int dmk_format::identify(util::random_read &io, uint32_t form_factor, const std:
 }
 
 
+uint32_t dmk_format::wide_fm(uint32_t val)
+{
+	uint32_t res = 0;
+	for (int i = 15; i >= 0; i--) {
+		if (val & (1 << i))
+			res |= (1 << (i*2 + 1));
+	}
+	return res;
+}
+
+uint32_t dmk_format::data_to_wide_fm(uint8_t val)
+{
+	uint16_t res = 0;
+	for (int i = 7; i >= 0; i--) {
+		res |= (1 << (i*2 + 1));	// clock
```

Comment:
Since this is just setting the clock bits to one, you could initialize `res` to `0xaaaa` instead.

Reply:
Changed in updated pull request.